File: crates/sl-frontend/src/lib/types/themes/store-page.ts
import * as TabsPrimitive from '@radix-ui/react-tabs';

export interface StoreCategorySelectorThemeProps
	extends React.ComponentProps<typeof TabsPrimitive.Root> {
	values?: string[];
	categoryListProps?: React.ComponentProps<typeof TabsPrimitive.List>;
	categoryTriggerProps?: React.ComponentProps<typeof TabsPrimitive.Trigger>;
}

export interface StoreCardThemeProps
	extends React.DetailedHTMLProps<
		React.HTMLAttributes<HTMLDivElement>,
		HTMLDivElement
	> {
	imageProps?: React.DetailedHTMLProps<
		React.ImgHTMLAttributes<HTMLImageElement>,
		HTMLImageElement
	>;
}

export interface StorePageThemeProps {
	containerProps?: React.DetailedHTMLProps<
		React.HTMLAttributes<HTMLDivElement>,
		HTMLDivElement
	>;
	innerContainerProps?: React.DetailedHTMLProps<
		React.HTMLAttributes<HTMLDivElement>,
		HTMLDivElement
	>;
	storeCategorySelector?: StoreCategorySelectorThemeProps;
	inputProps?: React.ComponentProps<'input'>;
}

File: crates/sl-core/src/launcher/themes/store_page.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct StoreCategorySelectorThemeProps {
    pub category_list_props: Option<Value>, 
    pub category_trigger_props: Option<Value>,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct StorePageThemeProps {
    pub container_props: Option<Value>,
    pub store_category_selector: Option<StoreCategorySelectorThemeProps>,
    pub input_props: Option<Value>,
}
